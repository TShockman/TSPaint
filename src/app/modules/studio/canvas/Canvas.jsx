import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import theme from './canvas.scss';
import ColorCell from '../colorCell';

export default class Canvas extends PureComponent {
  static propTypes = {
    colors: PropTypes.arrayOf(PropTypes.string).isRequired,
    colorCanvasCell: PropTypes.func.isRequired
  };

  render() {
    const {colors, colorCanvasCell} = this.props;
    return (
      <div className="canvas">
        {colors.map((color, i) => {
          return <ColorCell key={i} index={i} color={color} onSelect={colorCanvasCell}/>;
        })}
      </div>
    );
  }
}