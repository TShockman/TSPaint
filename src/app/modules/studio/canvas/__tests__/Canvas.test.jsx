import React from 'react';
import Canvas from '../index';
import ReactShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ReactShallowRenderer();

describe("Canvas", () => {
  it('renders consistently', () => {
    const props = {
      colorCanvasCell: jest.fn(),
      colors: ["#333333", "#F0F000", "#00F730", "#333333", "#F0F000", "#00F730"]
    };
    expect(renderer.render(<Canvas {...props}/>)).toMatchSnapshot();
  });
});