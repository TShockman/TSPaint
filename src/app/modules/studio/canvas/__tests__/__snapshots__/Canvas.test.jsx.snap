// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas renders consistently 1`] = `
<div
  className="canvas"
>
  <ColorCell
    color="#333333"
    index={0}
    onSelect={[MockFunction]}
  />
  <ColorCell
    color="#F0F000"
    index={1}
    onSelect={[MockFunction]}
  />
  <ColorCell
    color="#00F730"
    index={2}
    onSelect={[MockFunction]}
  />
  <ColorCell
    color="#333333"
    index={3}
    onSelect={[MockFunction]}
  />
  <ColorCell
    color="#F0F000"
    index={4}
    onSelect={[MockFunction]}
  />
  <ColorCell
    color="#00F730"
    index={5}
    onSelect={[MockFunction]}
  />
</div>
`;
